src/Fakta/Impl.fs

```diff
@@ -98,12 +110,16 @@ let getResponse (state : FaktaState) path (req : Request) =
   }
 
 let queryMeta dur (resp : Response) =
-  let headerFor key = resp.Headers |> Map.find (ResponseHeader.NonStandard key)
-  { lastIndex   = uint64 (headerFor "X-Consul-Index")
-    lastContact = Duration.FromSeconds (int64 (headerFor "X-Consul-Lastcontact"))
-    knownLeader = bool.Parse (headerFor "X-Consul-Knownleader")
+  let headerFor key = resp.Headers |> Map.tryFind (ResponseHeader.NonStandard key)
+  { lastIndex   = if (headerFor "X-Consul-Index").IsNone then UInt64.MinValue else  uint64 (headerFor "X-Consul-Index").Value
```

Comment:
Prefer module methods `Option.isNone` or in this case even better `headerFor "X-Consul-Index" |> Option.fold (fun s t -> uint64 t) UInt64.MinValue`